src/lib/blueprint.ts

```diff
@@ -47,10 +53,74 @@ interface Request {
   parameters: Parameter[]
 }
 
-interface Response {
+type Response = VoidResponse | ResourceResponse | ResourceListResponse
+
+interface BaseResponse {
   description: string
 }
 
+interface VoidResponse extends BaseResponse {
+  responseType: 'void'
+}
+
+interface ResourceResponse extends BaseResponse {
+  responseType: 'resource'
+  responseKey: string
+  resourceType: string
+}
+
+interface ResourceListResponse extends BaseResponse {
+  responseType: 'resource_list'
+  responseKey: string
+  resourceType: string
+}
+
+interface BaseProperty {
+  name: string
+  type: 'string' | 'enum' | 'record' | 'list' | 'object'
+  description?: string
+  isRequired?: boolean
+  isDeprecated?: boolean
+  deprecationMessage?: string
+}
+
+type Property =
+  | StringProperty
+  | EnumProperty
+  | RecordProperty
+  | ListProperty
+  | ObjectProperty
+
+interface StringProperty extends BaseProperty {
+  type: 'string'
+}
+
+interface EnumProperty extends BaseProperty {
+  type: 'enum'
+  values: EnumValue[]
+}
+
+interface EnumValue {
+  name: string
+  description: string
```

Comment:
We don't know how to generate this yet, so we can just remove it and add more metadata as it comes up.
```suggestion
```